### ISSUE TITLE
Should methods like MatrixClient.getRooms() throw a warning/error if client was not started?

### ISSUE THREAD

**Issue by cedricvanrompay** [matrix-org/matrix-js-sdk#904]
I lost quite some time with a bug that was caused by me calling `MatrixClient.createRoom` then `MatrixClient.getRooms` without having called `MatrixClient.startClient`
(as a result the client was not syncing and not updating its room list,
even though the call to `createRoom` did return successfully with a room Id).

I think that, unless there *are* some use cases where it makes sense to call `MatrixClient.getRooms` on a client that has not been started yet,
the method should probably throw an error or a warning if the client was not started.

This would apply to every method that requires the client to be synchronizing.

**Comment by turt2live** [comment]
I don't think it explicitly requires the client to be syncing: if the client was syncing, stored some data, and was re-loaded from that data then it should have rooms to report. Instead, clarifying the documentation on functions like this to say that they are more or less pointless without a syncing client could help.
